packages/google_identity_services_web/tool/run_tests.dart

```diff
@@ -0,0 +1,49 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// Runs `dart test -p chrome` in the root of the google_identity_services_web package.
+//
+// Called from the custom-tests CI action.
+//
+// usage: dart run tool/run_tests.dart
+// (needs a `chrome` executable in $PATH, or a tweak to dart_test.yaml)
+import 'dart:io';
+import 'package:path/path.dart' as p;
+
+Future<void> main(List<String> args) async {
+  final Directory workingDir =
+      Directory(p.dirname(Platform.script.path)).parent;
+
+  final int status = await _runProcess(
+    'dart',
+    <String>[
+      'test',
+      '-p',
+      'chrome',
```

Comment:
You'll need to replicate the plugin tooling logic to honor `CHROME_EXECUTABLE` here to make this hermetic, otherwise it'll be using the random current version of Chrome rather than the pinned Chromium.

Reply:
Isn't the `CHROME_EXECUTABLE` being set by `.cirrus.yml`? 

https://github.com/flutter/packages/blob/main/.cirrus.yml#L7-L19

I'm pretty sure that's pinning Chromium to [this version](https://github.com/flutter/packages/blob/main/script/install_chromium.sh#L14-L19).

Is this missing something (other than non-linux support)?

Reply:
Oh, does `flutter test` inherently use `CHROME_EXECUTABLE`? I thought we had to do it manually like [this](https://github.com/flutter/plugins/blob/2592df96f6ae2f1d2687de9fa4f9eae87e6e378d/script/tool/lib/src/drive_examples_command.dart#L117).

Reply:
Wowza, I need to check that! Now that you mention it, `flutter test` attempts to locate chrome in "expected locations", so it might be finding some other instance :/

I thought the images did NOT have any chrome installed, unless we did so!

Reply:
All right, so the way this seems to work is:

1. `dart test` expects `google-chrome` to be in the path in Linux. [Source](https://github.com/dart-lang/test/blob/5569fabbec6647f01f957b684e235125edbb67ec/pkgs/test/lib/src/runner/browser/default_settings.dart#L13).
2. We redefine that `executable` to `chrome` [here](https://github.com/flutter/packages/pull/2653/files#diff-dcff8596a2387124827f232ddb2e39c2c7d351c3593e2c54c86619ec0ea484e6R5).
3. `executable` seems to be only available on that yaml file. [Docs](https://github.com/dart-lang/test/blob/master/pkgs/test/doc/configuration.md#executable).

The above Works As Expected, just because we set the `$CHROME_DOWNLOAD_DIR/chrome-linux` directory into the $PATH, [here](https://github.com/flutter/packages/blob/main/.cirrus.yml#L14).

This is probably very brittle :)

I've asked (but received no answer yet) on how to pass the chrome executable as a command-line argument, so we don't have to set it in the $PATH (I don't see if it's even possible). Alternatively we could generate the `dart_test.yaml` file dynamically before running the test?

I'm going to create an issue to account for this tech debt, because `cross_file` has the same issue.

Reply:
https://github.com/flutter/flutter/issues/113079